### PR TITLE
Add stopWithFlush option to ffwd reporter & ffwd http reporter

### PR DESCRIPTION
**Pull request by lmuhlha**
Add stop with flush option for user's who would like to ensure a final flush of metrics when their service shuts down. 
Tests for:
- Use of stopWithFlush with no executor tick
- Ensure failure when there's no executor tick normally

### CODE REVIEW

File: ffwd-http-reporter/src/main/java/com/spotify/metrics/ffwdhttp/FastForwardHttpReporter.java [hunk 1]
```diff
@@ -440,6 +438,22 @@ public void stop() {
...
     }
 
+    public void stopWithFlush() {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(false);
+        }
+        if (executorOwner) {
+            executorService.shutdown();
+        }
+        //final flush of metrics
```
Comment (irock) [inline ffwd-http-reporter/src/main/java/com/spotify/metrics/ffwdhttp/FastForwardHttpReporter.java:448]: I wonder if the log line might be sufficient documentation for what's going on.

Comment (lmuhlha) [reply ffwd-http-reporter/src/main/java/com/spotify/metrics/ffwdhttp/FastForwardHttpReporter.java:448]: true haha

File: ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java [hunk 2]
```diff
@@ -218,4 +212,132 @@ public void shouldAddExtractedTags() throws Exception {
...
+    }
+
+    @Test
+    public void someReportingNoTickFails() {
+        doReturn(Observable.<Void>just(null)).when(httpClient).sendBatch(any(Batch.class));
+        fixedClock.setCurrentTime(TIME);
+        registry.counter(MetricId.build("counter"));
+        registry.derivingMeter(MetricId.build("deriving-meter"));
+
+
```
Comment (irock) [inline ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java:320]: nit: extra line break

File: ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java [hunk 3]
```diff
@@ -218,4 +212,132 @@ public void shouldAddExtractedTags() throws Exception {
...
+        for (final Batch b : batch.getAllValues()) {
+            assertEquals(commonTags, b.getCommonTags());
+            final Set<Batch.Point> points = new HashSet<>(b.getPoints());
+            points.removeAll(expected);
+            assertEquals("expected empty set of points", ImmutableSet.of(), points);
+        }
+    }
+
+    @Test
+    public void someReportingNoTickFails() {
```
Comment (irock) [inline ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java:314]: I'm not sure best description is that it fails? Perhaps a more descriptive name could be `noReportingIfNoTick`?

**Review by hexedpackets — APPROVED** [review]
I agree about renaming `someReportingNoTickFails`, otherwise LGTM

**Review by irock — APPROVED** [review]

**Review by malish8632 — APPROVED** [review]

**Review by hexedpackets — APPROVED** [review]

**Pull request merged**